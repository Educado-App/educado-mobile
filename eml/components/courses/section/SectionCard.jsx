import React, { useState } from 'react';
import { View, Text, Pressable, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { MaterialCommunityIcons } from "@expo/vector-icons";
import Collapsible from "react-native-collapsible";
import { useNavigation } from '@react-navigation/native'

export default function SectionCard({ sectionNumber, description, imageSrc, completed, total }) {
    const navigation = useNavigation();
    const isComplete = completed === total;
    const [isOpen, setIsOpen] = useState(false);


    const toggleDropdown = () => {
        setIsOpen(!isOpen);
    }

    const handleImagePress = () => {
        navigation.navigate('HomeStack'); // Replace with the name of the target screen
    }

    return (
        <View style={styles.shadowWrapper}>
            <Pressable onPress={toggleDropdown} style={styles.wrapper}>

                <View style={[styles.header, isComplete ? styles.headerComplete : {}]}>

                    <Text style={styles.title}>
                        Section {sectionNumber}
                    </Text>
                    <Text style={styles.completionText}>
                        {completed}/{total} completed
                    </Text>
                    <MaterialCommunityIcons
                        name={isOpen ? "chevron-up" : "chevron-down"}
                        size={25}
                        color="gray"
                    />
                </View>

                <Collapsible collapsed={!isOpen}>

                    <View style={styles.lineBreak} />
                    <Text style={styles.descriptionText}>{description}</Text>
                    <View style={styles.dropdownContent}>
                        <TouchableOpacity style={styles.imageContainer} onPress={handleImagePress}>
                            <Image source={imageSrc ? imageSrc : require('../../../assets/sectionThumbnail.png')} style={styles.image} />
                            <MaterialCommunityIcons name="play-circle-outline" size={50} color="white" style={styles.playIcon} />
                        </TouchableOpacity>
                    </View>
                </Collapsible>

            </Pressable>
        </View>
    );
}

const styles = StyleSheet.create({
    shadowWrapper: {
        backgroundColor: 'transparent',
        margin: 8,
        borderRadius: 10,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.3,
        shadowRadius: 4.65,
        elevation: 8,
    },
    wrapper: {
        backgroundColor: "#fff",
        borderRadius: 10,
        marginBottom: 15,
        marginHorizontal: 18,
        overflow: 'hidden',
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingHorizontal: 25,
        paddingVertical: 15,
    },
    title: {
        fontSize: 16,
        color: 'black',
        flex: 1,
    },
    dropdownContent: {
        flex: 1,
        width: '100%',
    },
    lineBreak: {
        height: 1,
        backgroundColor: '#e0e0e0',
    },
    imageContainer: {
        width: '100%',
        height: 300,
        alignItems: 'center',
        justifyContent: 'center',
        position: 'relative',
    },
    playIcon: {
        position: 'absolute',
        zIndex: 1,
        right: 10,
    },
    image: {
        width: '100%',
        height: 300,
        resizeMode: 'cover',
    },
    headerComplete: {
        backgroundColor: '#87CEEB', // Completion color - change to a green color if needed
    },
    completionText: {
        marginRight: 10,
        color: 'black',
    },
    descriptionText: {
        marginHorizontal: 20,
        marginVertical: 10
    },
});
