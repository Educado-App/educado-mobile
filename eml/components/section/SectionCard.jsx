import React, { useState } from 'react';
import { View, Pressable, Image, TouchableOpacity } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import Text from '../general/Text';
import Collapsible from 'react-native-collapsible';
import { useNavigation } from '@react-navigation/native';
import PropTypes from 'prop-types';


/**
 * A component that displays a section card with collapsible content.
 * @param {Object} section - The section object containing the section data.
 * @param {Object} course - The course object containing the course data.
 * @param {Number} progress - The progress containing the student's progress.
 * @returns {JSX.Element} - The SectionCard component.
 */
export default function SectionCard({ section, course, progress }) {


	const navigation = useNavigation();
	const isComplete = progress === section.components.length;
	const inProgress = 0 < progress && progress < section.components.length;
	const notPossible = progress > section.components.length;
	const [isOpen, setIsOpen] = useState(false);
	const backgroundColor = isComplete ? 'bg-limeGreenDarker' : inProgress ? 'bg-cyanBlue' : notPossible ? 'bg-error' : {};

	/**
     * Toggles the dropdown state.
     */
	const toggleDropdown = () => {
		setIsOpen(!isOpen);
	};


	/**
     * Handles the image press event.
     */
	const handleImagePress = () => {
		navigation.navigate('Components', {
			section: section,
			parsedCourse: course,
		});
	};


	return (
		<View>
			<Pressable testID="collapsible" onPress={toggleDropdown} className="bg-projectWhite rounded-lg shadow-lg shadow-opacity-[0.3] mb-[15] mx-[18] overflow-hidden elevation-[8]">
				<View className={'flex-row items-center justify-between px-[25] py-[15] ' + backgroundColor}>
					<Text className="text-[16px] font-bold text-projectBlack flex-[1]">
						{section.title}
					</Text>
					<Text className="mr-[10] text-projectBlack">
						{/* completed */}
						{progress}/{section.components.length} conclu√≠dos
					</Text>
					<MaterialCommunityIcons
						testID={isOpen ? 'chevron-up' : 'chevron-down'}
						name={isOpen ? 'chevron-up' : 'chevron-down'}
						size={25}
						color="gray"
					/>
				</View>

				<Collapsible collapsed={!isOpen}>
					<View className="h-[1] bg-disable" />
					<Text className="mx-[20] my-[10]">{section.description}</Text>
					<View className="w-[100%]">
						<TouchableOpacity className="w-[100%] h-[300] items-center justify-center relative"
							onPress={handleImagePress}>
							<Image source={require('../../assets/images/sectionThumbnail.png')} className="w-[100%] h-[300] object-cover" />
							<View className="absolute z-[1]">
								<MaterialCommunityIcons name="play-circle-outline" size={100} color="lightblue" />
							</View>
						</TouchableOpacity>
					</View>
				</Collapsible>
			</Pressable>
		</View>
	);
}

SectionCard.propTypes = {
	section: PropTypes.object,
	course: PropTypes.object,
	progress: PropTypes.number,
};