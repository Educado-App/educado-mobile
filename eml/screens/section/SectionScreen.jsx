import React, { useState, useEffect } from 'react';
import { Alert, View, TouchableOpacity } from 'react-native';
import Text from '../../components/general/Text';
import * as StorageService from '../../services/StorageService';
import SectionCard from '../../components/section/SectionCard';
import { ScrollView } from 'react-native-gesture-handler';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import CustomProgressBar from '../../components/exercise/Progressbar';
import BaseScreen from '../../components/general/BaseScreen';
import SubscriptionCancel from '../../components/section/CancelSubscriptionButton';
import { unsubscribe } from '../../services/StorageService';
import PropTypes from 'prop-types';
import { checkProgressCourse, checkProgressSection } from '../../services/utilityFunctions';

/**
 * Section screen component that displays a list of sections for a given course.
 * @param {object} route - The route object containing the courseId parameter.
 * @returns {JSX.Element} - The SectionScreen component.
 */
export default function SectionScreen({ route }) {
	SectionScreen.propTypes = {
		route: PropTypes.object,
	};
	const { course } = route.params;
	const navigation = useNavigation();
	const [sections, setSections] = useState(null);
	const [studentProgress, setStudentProgress] = useState(0);
	const [completedComponents, setCompletedComponents] = useState(0);

	/**
   * Loads the sections for the given course from the backend.
   * @param {string} id - The id of the course to load sections for.
   */
	async function loadSections(id) {
		const sectionData = await StorageService.getSectionList(id);
		setSections(sectionData);
	}

	const checkProgress = async () => {
		const progress = await checkProgressCourse(course.courseId);
		setStudentProgress(progress);
	};

	const checkProgressInSection = async (sectionId) => {
		const completed = await checkProgressSection(sectionId);
		setCompletedComponents(completed);
	};

	useEffect(() => {
		// this makes sure loadcourses is called when the screen is focused
		const update = navigation.addListener('focus', () => {
			checkProgress();
		});
		return update;
	}, [navigation]);

	// Fetch courses from backend and replace dummy data!
	useEffect(() => {
		let componentIsMounted = true;

		/**
     * Loads the sections and course data for the given courseId.
     */
		async function loadData() {
			await loadSections(course.courseId);
		}

		if (componentIsMounted) {
			loadData();
		}

		checkProgress();

		return () => componentIsMounted = false;
	}, []);

	/**
   * Displays an alert to confirm unsubscribing from the course.
   */
	const unsubAlert = () =>
		Alert.alert('Cancelar subscrição', 'Tem certeza?', [
			{
				text: 'Não',
				style: 'cancel',
			},
			{ text: 'Sim', onPress: () => { unsubscribe(course.courseId); setTimeout(() =>  {navigation.goBack();}, 300 ); }},
		]);

	return (
		<BaseScreen>
			<View className="flex flex-row flex-wrap items-center justify-between px-6 pt-[20%]">

				{/* Back Button */}
				<TouchableOpacity className="pr-3" onPress={() => navigation.goBack()}>
					<MaterialCommunityIcons name="chevron-left" size={25} color="black" />
				</TouchableOpacity>

				{/* Course Title */}
				<Text className="text-[25px] font-bold">{course.title}</Text>
			</View>

			{/* Conditionally render the sections if they exist */}
			{sections ? (
				sections.length === 0 ? null : (
					<View className="flex-[1] flex-col my-[10px]">

						{/* Progress Bar */}
						<CustomProgressBar width={60} progress={studentProgress} height={3}></CustomProgressBar>

						{/* Section Cards */}
						<ScrollView className="mt-[5%]" showsVerticalScrollIndicator={false}>
							{sections.map((section, i) => {
								checkProgressInSection(section.sectionId);
								return <SectionCard key={i} section={section} course={course} progress={completedComponents}></SectionCard>;
							})}
						</ScrollView>
						{/* Unsubscribe Button */}
						<SubscriptionCancel onPress={unsubAlert} />
					</View>
				)
			) : null}

		</BaseScreen>
	);
}
