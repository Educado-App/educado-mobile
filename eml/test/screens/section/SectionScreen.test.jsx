/**
 * This file contains the test suite for the SectionScreen component.
 * It tests the rendering of the component with sections loaded and the unsubAlert function.
 * @module SectionScreen.test
 */

import React from 'react';
import renderer from 'react-test-renderer';
import SectionScreen from '../../../screens/section/SectionScreen';
import SectionCard from '../../../components/section/SectionCard';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Alert } from 'react-native';
import { mockDataAsyncStorage } from '../../mockData/mockDataAsyncStorage';

/**
 * Mock navigation object used for testing.
 * @type {object}
 * @property {jest.Mock} goBack - Mock function for the goBack method.
 * @property {jest.Mock} navigate - Mock function for the navigate method.
 */
let mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
};

/**
 * Mocks the useNavigation hook from react-navigation/native.
 * @function useNavigation
 * @returns {object} - Returns the mock navigation object.
 */
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    mockNavigation,
    navigate: jest.fn(() => {
      navigated = true;
    }),
  }),
}));

jest.mock('../../../components/general/BaseScreen', () => {
  return jest.fn().mockImplementation(({ children }) => children);
});

// Mock the ToastNotification
jest.mock('../../../components/general/ToastNotification', () => {
  return jest.fn().mockImplementation(({ children }) => children);
});

/**
 * Flag indicating whether the navigate method was called.
 * @type {boolean}
 */
let navigated = false;

/**
 * The SectionScreen component being tested.
 * @type {ReactTestRenderer}
 */
let sectionScreen;

/**
 * Clears the AsyncStorage before each test.
 */
beforeEach(() => {
  navigated = false;
  AsyncStorage.clear();
});

/**
 * Resets the modules and restores all mocks after all tests have run.
 */
afterAll(() => {
  jest.resetModules();
  jest.restoreAllMocks();
});

describe('SectionScreen', () => {
  /**
   * Mock route object used for testing.
   * @type {object}
   * @property {object} params - Mock params object.
   * @property {number} params.courseId - Mock courseId parameter.
   */
  const route = {
    params: {
      course: {
        courseId: 1,
        title: 1
      },
    },
  };

  /**
   * Mock sections array used for testing.
   * @type {Array<object>}
   */
  const mockData = mockDataAsyncStorage();

  /**
   * Tests the rendering of the SectionScreen component with sections loaded.
   */
  
  /*it('Renders section screen with sections loaded', async () => {
    const StorageService = require('../../../services/StorageService');
    jest.spyOn(StorageService, 'getSectionList').mockResolvedValue(mockData.sectionData);


    await renderer.act(async () => {
      return sectionScreen = renderer.create(<SectionScreen route={route} />);
    });


    await new Promise((resolve) => setTimeout(resolve, 0));

    expect(await sectionScreen.root.findAllByType(SectionCard)).toHaveLength(2);
    expect(await sectionScreen.toJSON()).toMatchSnapshot();
  });
  */

  describe('unsubAlert', () => {
    /**
     * Tests that the unsub function is called when "Sim" is pressed.
     */
    it('should call unsub function when \'Sim\' is pressed', async () => {
      /**
       * Function that displays an alert asking the user if they want to unsubscribe.
       * @function unsubAlert
       * @returns {void}
       */
      const unsubAlert = () => {
        Alert.alert('Cancelar subscrição', 'Tem certeza?', [
          {
            text: 'Não',
            onPress: () => console.log('No Pressed'),
            style: 'cancel',
          },
          { text: 'Sim', onPress: unsub },
        ]);
      };
      const unsub = jest.fn();
      const alert = jest.spyOn(Alert, 'alert');
      alert.mockImplementation((_, __, buttons) => buttons[1].onPress());
      unsubAlert();
      expect(alert).toHaveBeenCalled();
      expect(unsub).toHaveBeenCalled();
      alert.mockRestore();
    });

    /**
     * Tests that the unsub function is not called when "Não" is pressed.
     */
    it('should not call unsub function when \'Não\' is pressed', async () => {
      /**
       * Function that displays an alert asking the user if they want to unsubscribe.
       * @function unsubAlert
       * @returns {void}
       */
      const unsubAlert = () => {
        Alert.alert('Cancelar subscrição', 'Tem certeza?', [
          {
            text: 'Não',
            onPress: () => console.log('No Pressed'),
            style: 'cancel',
          },
          { text: 'Sim', onPress: unsub },
        ]);
      };
      const unsub = jest.fn();
      const alert = jest.spyOn(Alert, 'alert');
      alert.mockImplementation((_, __, buttons) => buttons[0].onPress());
      unsubAlert();
      expect(alert).toHaveBeenCalled();
      expect(unsub).not.toHaveBeenCalled();
      alert.mockRestore();
    });
  });
});