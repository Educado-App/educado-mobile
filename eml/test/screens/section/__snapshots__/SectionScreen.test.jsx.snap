// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionScreen Renders section screen with sections loaded 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#f1f9fb",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "paddingLeft": 24,
          "paddingRight": 24,
        },
        {
          "paddingTop": "20%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "paddingRight": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰅁
      </Text>
    </View>
    <Text
      style={
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          [
            {
              "fontSize": 25,
            },
            {
              "fontWeight": "700",
            },
          ],
        ]
      }
    />
    <View
      style={
        {
          "flex": 1,
        }
      }
    />
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "alignItems": "flex-end",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 16,
          }
        }
        testID="subscriptionCancelButton"
      >
        <Text
          style={
            [
              {
                "textAlign": "center",
              },
              {
                "fontFamily": "Montserrat-Bold",
              },
              {
                "fontSize": 16,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "black",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰃀
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "flexDirection": "column",
        },
        {
          "marginBottom": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "justifyContent": "space-around",
          },
        ]
      }
    >
      <View
        animated={true}
        animationConfig={
          {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={false}
        indeterminateAnimationDuration={1000}
        onLayout={[Function]}
        progress={0.5}
        style={
          [
            {
              "backgroundColor": "rgba(228, 242, 245, 1)",
              "borderColor": "rgba(94, 204, 233, 1)",
              "borderRadius": 8,
              "borderWidth": 0,
              "overflow": "hidden",
              "width": 450,
            },
            undefined,
          ]
        }
        useNativeDriver={false}
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(94, 204, 233, 1)",
              "height": 40.019999999999996,
              "transform": [
                {
                  "translateX": 0,
                },
                {
                  "translateX": -112.5,
                },
                {
                  "scaleX": 0.50005,
                },
              ],
            }
          }
        />
      </View>
      <Text
        style={
          [
            {
              "paddingLeft": 20,
              "paddingRight": 20,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Montserrat-Bold",
            },
            {
              "fontSize": 12,
            },
            {
              "color": "#383838",
            },
          ]
        }
      >
        50
        %
      </Text>
    </View>
    <RCTScrollView
      collapsable={false}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      showsVerticalScrollIndicator={false}
      style={
        [
          {
            "marginTop": "5%",
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "marginBottom": "2.2%",
                "marginLeft": "2.2%",
                "marginRight": "2.2%",
                "marginTop": "2.2%",
              },
              {
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                },
                {
                  "marginBottom": 15,
                },
                {
                  "marginLeft": 18,
                  "marginRight": 18,
                },
                {
                  "overflow": "hidden",
                },
                {
                  "elevation": 8,
                },
              ]
            }
            testID="collapsible"
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "paddingLeft": 25,
                    "paddingRight": 25,
                  },
                  {
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "fontSize": 16,
                      },
                      {
                        "fontWeight": "700",
                      },
                      {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ],
                  ]
                }
              >
                Section 1
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginRight": 10,
                      },
                    ],
                  ]
                }
              >
                0
                /
                10
                 concluídos
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="chevron-down"
              >
                󰅀
              </Text>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      {
                        "height": 1,
                      },
                      {
                        "backgroundColor": "#e0e0e0",
                      },
                    ]
                  }
                />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      [
                        {
                          "marginLeft": 20,
                          "marginRight": 20,
                        },
                        {
                          "marginBottom": 10,
                          "marginTop": 10,
                        },
                      ],
                    ]
                  }
                >
                  Description for section 1
                </Text>
                <View
                  style={
                    [
                      {
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "height": 300,
                        "justifyContent": "center",
                        "opacity": 1,
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        [
                          {
                            "width": "100%",
                          },
                          {
                            "height": 300,
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        [
                          {
                            "position": "absolute",
                          },
                          {
                            "zIndex": 1,
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "lightblue",
                              "fontSize": 100,
                            },
                            undefined,
                            {
                              "fontFamily": "material-community",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        󰐍
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "marginBottom": "2.2%",
                "marginLeft": "2.2%",
                "marginRight": "2.2%",
                "marginTop": "2.2%",
              },
              {
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                },
                {
                  "marginBottom": 15,
                },
                {
                  "marginLeft": 18,
                  "marginRight": 18,
                },
                {
                  "overflow": "hidden",
                },
                {
                  "elevation": 8,
                },
              ]
            }
            testID="collapsible"
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "paddingLeft": 25,
                    "paddingRight": 25,
                  },
                  {
                    "paddingBottom": 15,
                    "paddingTop": 15,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "fontSize": 16,
                      },
                      {
                        "fontWeight": "700",
                      },
                      {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ],
                  ]
                }
              >
                Section 2
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginRight": 10,
                      },
                    ],
                  ]
                }
              >
                0
                /
                10
                 concluídos
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
                testID="chevron-down"
              >
                󰅀
              </Text>
            </View>
            <View
              collapsable={false}
              pointerEvents="none"
              style={
                {
                  "height": 0,
                  "overflow": "hidden",
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                onLayout={[Function]}
                style={{}}
              >
                <View
                  style={
                    [
                      {
                        "height": 1,
                      },
                      {
                        "backgroundColor": "#e0e0e0",
                      },
                    ]
                  }
                />
                <Text
                  style={
                    [
                      {
                        "fontFamily": "Montserrat-Regular",
                      },
                      [
                        {
                          "marginLeft": 20,
                          "marginRight": 20,
                        },
                        {
                          "marginBottom": 10,
                          "marginTop": 10,
                        },
                      ],
                    ]
                  }
                >
                  Description for section 2
                </Text>
                <View
                  style={
                    [
                      {
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "height": 300,
                        "justifyContent": "center",
                        "opacity": 1,
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                  >
                    <Image
                      source={1}
                      style={
                        [
                          {
                            "width": "100%",
                          },
                          {
                            "height": 300,
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        [
                          {
                            "position": "absolute",
                          },
                          {
                            "zIndex": 1,
                          },
                        ]
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "lightblue",
                              "fontSize": 100,
                            },
                            undefined,
                            {
                              "fontFamily": "material-community",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                      >
                        󰐍
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
